From e3b2078f732af40ad351416262b6e3b4ecd788e5 Mon Sep 17 00:00:00 2001
From: Cai YiWei <cyw@rock-chips.com>
Date: Sat, 30 Jan 2021 11:11:34 +0800
Subject: [PATCH] media: rockchip: isp: reorder of subdev stream

on: isp->phy->sensor(start to output)
off: sensor(stop already)->phy->isp

Change-Id: I592f8e8e36f4c912e3f38b3f4266e98bfd9e56e4
Signed-off-by: Cai YiWei <cyw@rock-chips.com>
---
 drivers/media/platform/rockchip/isp/dev.c | 20 ++++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/drivers/media/platform/rockchip/isp/dev.c b/drivers/media/platform/rockchip/isp/dev.c
index 4cf1931e1bdc..d297a4dc29f7 100644
--- a/drivers/media/platform/rockchip/isp/dev.c
+++ b/drivers/media/platform/rockchip/isp/dev.c
@@ -274,16 +274,16 @@ static int rkisp_pipeline_set_stream(struct rkisp_pipeline *p, bool on)
 			enable_irq(dev->vs_irq);
 		rockchip_set_system_status(SYS_STATUS_ISP);
 		v4l2_subdev_call(&dev->isp_sdev.sd, video, s_stream, true);
-	}
-
-	/* phy -> sensor */
-	for (i = 0; i < p->num_subdevs; ++i) {
-		ret = v4l2_subdev_call(p->subdevs[i], video, s_stream, on);
-		if (on && ret < 0 && ret != -ENOIOCTLCMD && ret != -ENODEV)
-			goto err_stream_off;
-	}
-
-	if (!on) {
+		/* phy -> sensor */
+		for (i = 0; i < p->num_subdevs; ++i) {
+			ret = v4l2_subdev_call(p->subdevs[i], video, s_stream, on);
+			if (on && ret < 0 && ret != -ENOIOCTLCMD && ret != -ENODEV)
+				goto err_stream_off;
+		}
+	} else {
+		/* sensor -> phy */
+		for (i = p->num_subdevs - 1; i >= 0; --i)
+			v4l2_subdev_call(p->subdevs[i], video, s_stream, on);
 		if (dev->vs_irq >= 0)
 			disable_irq(dev->vs_irq);
 		v4l2_subdev_call(&dev->isp_sdev.sd, video, s_stream, false);
-- 
2.35.3

